staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -94,12 +95,20 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 		}
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		var transformers []rest.TransformFunc
-		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && mutatingAdmission.Handles(admission.Update) {
+		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
 			transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
-				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+				if mutatingAdmission.Handles(admission.Create) {
+					if accessor, err := meta.Accessor(oldObj); err == nil && accessor.GetResourceVersion() == "" {
```

Comment:
I think this check should be the outer one, and only one of the checks ```mutatingAdmission.Handles(admission.Create)```/```mutatingAdmission.Handles(admission.Update)``` should be done, depending on the result of this ```accessor.GetResourceVersion() == ""``` check.

The way you have it now doesn't work in the case where ```mutatingAdmission.Handles(admission.Create)``` is false, ```accessor.GetResourceVersion() == ""``` is true, and ```mutatingAdmission.Handles(admission.Update)``` is true

In that case it would call Update admission, but it should be calling nothing.

Reply:
Nice catch :) Thx

Reply:
Nice catch :) Thx